src/components/recordset/table-row.tsx

```diff
@@ -153,30 +156,59 @@ const TableRow = ({
     if (sensor && rowValues.length === 1 && justOverflows.length === 0) sensor.detach();
   }
 
-  // TODO: This assumes that tuple is set before rowValues. And that useEffect triggers before useLayoutEffect
+  // This assumes that tuple is set before rowValues. And that useEffect triggers before useLayoutEffect
   // NOTE: if the tuple changes, the table-row component isn't destroyed so the overflows need to be reset
   useEffect(() => {
     if (disableMaxRowHeightFeature) return;
     setOverflow([]);
   }, [tuple]);
 
+  // attach resize sensor to the table row and an onload event to each <img> tag in the table row
+  // onload will update a state variable to communicate when all images have loaded to trigger overflow logic once more
   useLayoutEffect(() => {
     if (!rowContainer.current || disableMaxRowHeightFeature) return;
     const tempSensor = new ResizeSensor(
       rowContainer.current,
-      () => setTimeout(() => {
-        initializeOverflows()
-        // images take ~200ms to load so wait at least that long before running
-      }, 200)
+      () => initializeOverflows()
     )
 
     setSensor(tempSensor);
+    
+    // fetch all <img> tags with -chaise-post-load class and keep count of the total
+    // attach an onload function that updates how many have loaded
+    const imgTags = rowContainer.current.querySelectorAll('img.-chaise-post-load');
```

Comment:
It seems like we're not consistent in how we're attaching this class in ermrestjs:

- If you use the `![]()` syntax, we're attaching the class to the `img` tag like the selector that you have.
- If you use the `:::image` syntax, we're attaching the class to `figure` element that is wrapping the image.

So I think you should change this selector to `img.-chaise-post-load, .-chaise-post-load img`.

And let's not hardcode `-chaise-post-load` here and instead add it to `CLASS_NAMES` that we have in the `constants.ts`.